code/processes/eodprocess.q

```diff
@@ -0,0 +1,90 @@
+\d .eod
+
+taildir:hsym `$getenv`KDBTAIL;                                             /-load in taildir env variables
+hdbdir:hsym `$getenv`KDBHDB;                                               /-load in hdb env variables
+ignorelist:@[value;`ignorelist;`heartbeat`logmsg];                         /-list of tables to ignore
+taildbs:key taildir;                                                       /-list of tailDBs that need saved to HDB
+taildirs:();                                                               /-empty list to append tailDB paths to - to be used
+                                                                           / when HDB save is complete to delete tailDB partitions
+\d .
+
+savescompleted:0;                                                          /-variable to check how many tailDBs have been saved to HDB
+
+endofday:{[pt;procname]
+  /- function to trigger data load & save to HDB once endofday message is received from tailer(s)
+  .lg.o[`endofday;"end of day message received from ",string[procname]," - ",string[pt]];
+  loadandsave[pt;procname];
+  };
+
+loadandsave:{[pt;procname]
+  /- function to load TDB data and save data to HDB
+  taildir:`$1_string ` sv (.eod.taildir;procname;`$string pt);
+  .eod.taildirs,:taildir;
+  .lg.o[`load;"loading TDB data from ",string[procname]];
+  @[.Q.l; taildir; {[e] .lg.e[`load;"Failed to load TDB : ",e]}];
+
+  /-define list of tables to be saved to HDB
+  tablelist:tables[] except .eod.ignorelist;
+
+  /-save tables to HDB
+  savetables[.eod.hdbdir;pt;tablelist];
+  savescompleted+::1;
```

Comment:
So just to check, this saves completed gets updated each time one of the tailers triggers the eod sort and once all tailers have triggered the eod sort the savecompleted runs to delete directories and reset the variables?

Reply:
Yes, so `savescompleted` acts as a counter as we expect the same number of saves as we have tailDBs, so it gets updated every save and once all tailDBs have been saved it'll trigger the `savecomplete` function to add any attributes and delete the tailDB directories. And yeah the variables are reset within `savecomplete` too so the the next day's save can run without issues.
We discussed in the demo this counter approach probably isn't the best method as there could be some issues if something needs saved twice or something, but I think Gary said he'd be happy enough with it as a starting point

Reply:
Okay great. Yeah that makes sense, worth getting it working for now at least.